# CVE-2021-28861


## Metadata
| Published | Last Modified | Assigner | Version | Format |
| --------- | ------------- | -------- | ------- | ------ |
| 2022-08-23 | 2022-09-22 | cve@mitre.org | 4.0 | MITRE | 

## Description
| Language | Description |
| -------- | ----------- |
| en | "** DISPUTED ** Python 3.x through 3.10 has an open redirection vulnerability in lib/http/server.py due to no protection against multiple (/) at the beginning of URI path which may leads to information disclosure. NOTE: this is disputed by a third party because the http.server.html documentation page states ""Warning: http.server is not recommended for production. It only implements basic security checks.""" |

## Configurations
| Vulnerable | CPE 22 URI | CPE 23 URI | CPE Name | Version Start Including | Version End Including | Version Start Excluding | Version End Excluding | Operator |
| ---------- | ---------- | ---------- | -------- | ----------------------- | --------------------- | ----------------------- | --------------------- | -------- |
| True  cpe:2.3:a:python:python:*:*:*:*:*:*:*:* | [] | 3.0   | 3.10.1 | OR |

## CVSSv3 Information
| Score | Severity | Impact Score | Exploitability Score | Vector String | Version | Attack Vector | Attack Complexity | Privileges Required | User Interaction | Scope | Confidentiality Impact | Integrity Impact | Availability Impact |
| ----- | -------- | ------------ | -------------------- | ------------- | ------- | ------------- | ----------------- | ------------------- | ---------------- | ----- | ---------------------- | ---------------- | ------------------- |
| 7.4 | HIGH | 4.0 | 2.8 | CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:H/I:N/A:N | 3.1 | NETWORK | LOW | NONE | REQUIRED | CHANGED | HIGH | NONE | NONE |

## CVSSv2 Information
*** Not Provided/Calculated ***

## Problem Type
CWE-601

## References
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
https://github.com/python/cpython/pull/93879|https://github.com/python/cpython/pull/93879|MISC|['Patch', 'Third Party Advisory']
https://bugs.python.org/issue43223|https://bugs.python.org/issue43223|MISC|['Issue Tracking', 'Vendor Advisory']
https://github.com/python/cpython/pull/24848|https://github.com/python/cpython/pull/24848|MISC|['Patch', 'Third Party Advisory']
FEDORA-2022-f511f8f58b|https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/OKYE2DOI2X7WZXAWTQJZAXYIWM37HDCY/|FEDORA|[]
